aws/resource_aws_accessanalyzer_analyzer.go

```diff
@@ -33,10 +34,13 @@ func resourceAwsAccessAnalyzerAnalyzer() *schema.Resource {
 
 		Schema: map[string]*schema.Schema{
 			"analyzer_name": {
-				Type:         schema.TypeString,
-				Required:     true,
-				ForceNew:     true,
-				ValidateFunc: validation.NoZeroValues,
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 255),
+					validation.StringMatch(regexp.MustCompile(`^[A-Za-z][A-Za-z0-9_.-]*$`), ""),
```

Comment:
Looks like I should write up the linter for this, but its generally better for practitioners to have a more consumable validation failure message:

```suggestion
					validation.StringMatch(regexp.MustCompile(`^[A-Za-z][A-Za-z0-9_.-]*$`), "must begin with a letter and contain only alphanumeric, underscore, period, or hyphen characters"),
```

Or as we do in a few other places for readability:

```go
					validation.StringMatch(regexp.MustCompile(`^[A-Za-z]`), "must begin with a letter"),
					validation.StringMatch(regexp.MustCompile(`^[A-Za-z0-9_.-]*$`), "must contain only alphanumeric, underscore, period, or hyphen characters"),
```